src/hooks/index.ts

```diff
@@ -971,3 +940,65 @@ export const useBalances = (wallet: YoroiWallet): YoroiAmounts => {
 
   return Utxos.toAmounts(utxos, primaryTokenId)
 }
+
+export const AUTH_METHOD_KEY = '/appSettings/authMethod'
+export const useAuthMethod = (storage: Storage, options?: UseQueryOptions<AuthMethodState, Error>) => {
+  const query = useQuery({
+    suspense: true,
+    queryKey: ['authMethod'],
+    queryFn: async () => {
+      const authMethod: AuthMethod | '' = await Promise.resolve(AUTH_METHOD_KEY)
+        .then(storage.getItem)
+        .then((storedAuthMethod) => (!isEmptyString(storedAuthMethod) ? storedAuthMethod : '""'))
+        .then(JSON.parse)
```

Comment:
define a custom parse function that guarantees valid data
```
const parseAuthMethod = (data: any): data is 'os' | 'pin' | undefined => {
  if (!data) return undefined
  try {
    const parsed = JSON.parse(data)
    // return valid data
  } catch (error) {
    // return valid data
  }
  
}
```

Reply:
if (!data) return true?